src/bin/ion/commands/beta/generate/templates/rust/struct.templ

```diff
@@ -76,23 +83,24 @@ impl {{ target_kind_name }} {
         {% if abstract_data_type == "Value" %}
             {% for field in fields %}
                 {% if field.value | is_built_in_type ==false  %}
-                    self.{{ field.name }}.write_to(writer)?;
+                    self.{{ field.name | snake }}.write_to(writer)?;
                 {% else %}
                     writer.write_{{ field.value | lower }}(self.value)?;
                 {% endif %}
             {% endfor %}
-        {% elif abstract_data_type == "Struct"%}
+        {% elif abstract_data_type is object and abstract_data_type | get(key="Structure") %}
             writer.step_in(IonType::Struct)?;
             {% for field in fields %}
             writer.set_field_name("{{ field.name }}");
                 {% if field.value | is_built_in_type == false %}
-                    self.{{ field.name }}.write_to(writer)?;
+                    self.{{ field.name | snake }}.write_to(writer)?;
                 {% else %}
                     writer.write_{{ field.value | lower }}(self.{{ field.name }})?;
+                    writer.write_{{ field.value | lower }}(self.{{ field.name | snake }}.to_owned())?;
```

Comment:
This is pointing to `ion-rs` writer API rather then the generated writer API which has the naming `write_to(VALUE)`. But this is required because write_i64, write_f32, write_f64, write_bool requires an owned value. 
There si an alternative solution possible where I can check the Ion type of value being written and if its f64,f32,i64, bool then use `to_owned` or `*`.  
Also, this is subject to change once we have lazy reader and writer replaced in these APIs.

Reply:
Ok. Let's leave it for now, but make an issue or a note somewhere to fix this before it is released.